src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxTreeFactoryService.cs

```diff
@@ -45,10 +45,10 @@ public override SyntaxTree CreateSyntaxTree(string fileName, ParseOptions option
                 return SyntaxFactory.SyntaxTree(root, options, fileName, encoding);
             }
 
-            public override SyntaxTree ParseSyntaxTree(string fileName, ParseOptions options, SourceText text, CancellationToken cancellationToken)
+            public override SyntaxTree ParseSyntaxTree(string filePath, ParseOptions options, SourceText text, ImmutableDictionary<string, ReportDiagnostic> treeDiagnosticReportingOptionsOpt, CancellationToken cancellationToken)
```

Comment:
So that's not the right one either, since that's using the [wrong comparer](https://github.com/dotnet/roslyn/blob/e14e5f559ecbaa13424ede01296e06b54ba6b224/src/CodeStyle/Core/Analyzers/AnalyzerConfigOptions.cs#L10-L14). A comparer for an empty dictionary, you say? Well, yes, it matters if somebody grabs it and starts adding keys into it. The compiler is precise about this, and rather than duplicating the precision I'm just letting the optionalness of this flow through.

Reply:
💡 Consider using a predefined instance somewhere with the correct comparer and simplifying these APIs.

Reply:
Tagging @agocke for the API feedback.